spec/faraday/rack_builder_spec.rb

```diff
@@ -193,4 +193,29 @@ class Broken < Faraday::Middleware
       end
     end
   end
+
+  context 'when middleware is added with arguments' do
+    let(:conn) { Faraday::Connection.new {} }
+
+    class Dog < Faraday::Middleware
+      attr_accessor :name
+      def initialize(app, name)
+        super(app)
+        @name = name
+      end
+    end
+    class Cat < Faraday::Middleware
+      attr_accessor :name
+      def initialize(app, name:)
+        super(app)
+        @name = name
+      end
+    end
```

Comment:
RSpec is powerful, we can inline these into not-accessible anywhere else variables, easily:


```suggestion
    let(:dog_middleware_class) do
      Class.new(Faraday::Middleware) do
        attr_accessor :name
        def initialize(app, name)
          super(app)
          @name = name
        end
      end
    end

    let(:cat_middleware_class) do
      Class.new(Faraday::Middleware) do
        attr_accessor :name
        def initialize(app, name:)
          super(app)
          @name = name
        end
      end
    end
```

And then use those in place of the `Dog` and `Cat`.